src/solc_0.8/claims/MultiGiveaway/MultiGiveaway.sol

```diff
@@ -186,15 +196,4 @@ contract MultiGiveaway is AccessControl, ClaimERC1155ERC721ERC20, ERC2771Handler
     function _msgData() internal view override(Context, ERC2771Handler) returns (bytes calldata) {
         return ERC2771Handler._msgData();
     }
-
-    /// @dev Prevent the renunciation of the DEFAULT_ADMIN_ROLE
-    /// @param role the role to revoke
-    /// @param account the address being revoked
-    function _revokeRole(bytes32 role, address account) internal virtual override {
-        if (role == DEFAULT_ADMIN_ROLE) {
-            revert("MULTIGIVEAWAY_PREVENT_REVOKE");
-        }
-
-        super._revokeRole(role, account);
-    }
```

Comment:
We should prevent the revoke for the  admin role. Why did you remove this ?

Reply:
I thought we agreed it was not needed:

DEFAULT_ADMIN_ROLE is a multisig and cannot be removed accidentally

DEFAULT_ADMIN_ROLE can re-add the MULTIGIVEAWAY_ROLE is it is accidentally removed

We also want to retain the ability to delete corrupted accounts for MULTIGIVEAWAY_ROLE

Reply:
Just re-read this. Makes sense to have it for the MULTIGIVEAWAY_ROLE

Reply:
As discussed, we leave this code out + we accept the risk, however we cannot accidentally remove DEFAULT_ADMIN_ROLE because it's a multisig.